### PR TITLE
fix: Add tests for functions identifying if messages are already submitted

### PR DESCRIPTION
**Pull request by ameten**
### Description

Add tests for functions identifying if messages are already submitted

### Related issues

- Contributing - https://linear.app/hyperlane-xyz/issue/ENG-2457/lander-tx-submission-observability-and-alerting-fix

### Backward compatibility

Yes

### Testing

More unit tests

## Summary by CodeRabbit

* **Refactor**
  * Enhanced relayer message processor abstraction patterns to support more flexible component implementations.

* **Tests**
  * Added comprehensive test coverage for operation confirmation and submission status checking, including edge cases and error scenarios.

### CODE REVIEW

File: rust/main/agents/relayer/src/msg/message_processor/tests/tests_has_operation_been_submitted.rs [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+
+    let op = Box::new(MockQueueOperation { id: message_id }) as QueueOperation;
+
+    let result = has_operation_been_submitted(
+        Arc::new(mock_entrypoint) as Arc<dyn Entrypoint + Send + Sync>,
+        Arc::new(mock_db) as Arc<dyn HyperlaneDb>,
+        &op,
+    )
+    .await;
+
```
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/agents/relayer/src/msg/message_processor/tests/tests_has_operation_been_submitted.rs]: We'll keep using `as` because it works since Rust 1.45+
